File: JINSEOP/프로그래머스_WEEK10/올바른 괄호.cpp
#include <string>
#include <stack>
using namespace std;

stack<char> validator;
bool check_parenth(string s) {
    if (s[0] == ')') return false;

    for (auto c : s) {
        if (c == '(') validator.push(c);
        else {
            if (validator.size() == 0) return false;
            else if (validator.top() == '(') validator.pop();
        }
    }

    if (!validator.empty()) return false;
    else return true;
}

bool solution(string s){
    bool answer = true;

    answer = check_parenth(s);
    return answer;
}

File: DONGUK/프로그래머스_WEEK_10/올바른 괄호.py
def check(li):
    stack = []

    for data in li:
        if data == '(': 
            stack.append(data)

        else: 
            if len(stack) == 0:
                return False
            
            stack.pop()
            
    
    if len(stack) == 0:
        return True
    
    else:
        return False
            
        

def solution(s):
    answer = check(s)
    return answer